### PR TITLE
Code samples READMEs

### PR DESCRIPTION
**Pull request by hsz**

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Build, Test, Mirror
+name: Build
```
Comment (YannCebron) [inline .github/workflows/build-test.yml:1]: accident?

File: .idea/compiler.xml [hunk 2]
```diff
@@ -35,8 +35,6 @@
       <module name="module" target="1.8" />
```
Comment (YannCebron) [inline .idea/compiler.xml:35]: accident?

File: code_samples/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+| [Product Specific - PyCharm Sample](./product_specific/pycharm_basics) | Plugin project configuration for the PyCharm IDE.                                                                                                      |
+| [Project Model](./project_model)                                       | Interacts with the project model, adds menu items to `Tools` and `Editor Context` menus.                                                               |
+| [Project View Pane](./project_view_pane)                               | Project View Pane listing only image files.                                                                                                            |
+| [Project Wizard](./project_wizard)                                     | Project Wizard example with demo steps.                                                                                                                |
+| [Psi Demo](./psi_demo)                                                 | PSI Navigation features presentation.                                                                                                                  |
```
Comment (YannCebron) [inline code_samples/README.md:34]: psi->PSI

File: code_samples/action_basics/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+Action Sample Project
+plugin depends on the [IntelliJ Platform SDK][docs] and [Gradle][docs:gradle] as a build system.
+
+The main plugin definition file is stored in the [plugin.xml][file:plugin.xml] file, which is created accordingly
+to the [Plugin Configuration File documentation][docs:plugin.xml]. It describes definitions of the actions, extensions,
+or listeners provided by the plugin.
+
+### Actions
+
+| Name                                               | Implementation Class                                      | Interface                      |
```
Comment (YannCebron) [inline code_samples/action_basics/README.md:30]: Name = ID?

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (hsz) [reply .github/workflows/build-test.yml:1]: Nope, it's just a name - also used in Badge images. I want to keep it short.

File: .idea/compiler.xml (hunk 2 above)
Comment (hsz) [reply .idea/compiler.xml:35]: Will it break anything?

Comment (YannCebron) [reply .idea/compiler.xml:35]: there should be no change

**Comment by hsz** [comment]
@YannCebron Everything is resolved.

File: code_samples/product_specific/pycharm_basics/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# PyCharm Sample [![JetBrains IntelliJ Platform SDK Docs](https://jb.gg/badges/docs.svg)][docs]
+*Reference: [PyCharm Plugin Development in IntelliJ SDK Docs][docs:pycharm]*
+
+## Quickstart
+
+PyCharm Sample is a plugin that depends on the PyCharm IDE having the proper dependencies specified
+in the Gradle configuration file. Implementation utilizes a simple action added to the *MainMenu* group displaying
```
Comment (JohnHake) [inline code_samples/product_specific/pycharm_basics/README.md:7]: Suggest: _The PyCharm Sample plugin demonstrates the correct Gradle project configuration for developing plugins targeting the PyCharm IDE. A simple action class is implemented to demonstrate adding an entry to the PyCharm IDE *MainMenu* group._

File: code_samples/project_model/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+# Project Model Sample [![JetBrains IntelliJ Platform SDK Docs](https://jb.gg/badges/docs.svg)][docs]
+*Reference: [PyCharm Plugin Development in IntelliJ SDK Docs][docs:pycharm]*
+
+## Quickstart
+
+Project Model Sample project provides five actions that present data extracted using `ProjectRootManager` instance
```
Comment (JohnHake) [inline code_samples/project_model/README.md:6]: Suggest: _The Project Model Sample project..._ 
and _rename the SDK used in the project,_

File: code_samples/project_view_pane/README.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# Project View Pane Demo [![JetBrains IntelliJ Platform SDK Docs](https://jb.gg/badges/docs.svg)][docs]
+*Reference: [Plugin Services in IntelliJ SDK Docs][docs:project_view]*
+
+## Quickstart
+
+The current demo describes an implementation of the `projectViewPane` extension point, which allows to create
+an additional presentation type for the Project view pane. `ImagesProjectViewPane` limits the project tree to the images
+only.
```
Comment (JohnHake) [inline code_samples/project_view_pane/README.md:8]: Suggest:_The Project View Pane plugin has an implementation of the `projectViewPane` extension point, which supports creating new presentation types for the Project view pane. The `ImagesProjectViewPane` implementation in this code sample limits the project tree to only images._

File: code_samples/psi_demo/README.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
+# PSI Demo [![JetBrains IntelliJ Platform SDK Docs](https://jb.gg/badges/docs.svg)][docs]
+*Reference: [Navigating the PSI in IntelliJ SDK Docs][docs:navigating_psi]*
+
+## Quickstart
+
+PSI Demo project demonstrates working with the PSI Navigation by implementing `AnAction` that through the message
+dialog, informs about:
```
Comment (JohnHake) [inline code_samples/psi_demo/README.md:7]: Suggest: _The PSI Demo project shows exploring PSI Navigation by implementing `AnAction` that explores a `PsiFile` associated with a file open in the editor. Using the plugin's action and message dialog, a user can move the editor caret and see information about:_

File: code_samples/settings/README.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
+# Settings Example [![JetBrains IntelliJ Platform SDK Docs](https://jb.gg/badges/docs.svg)][docs]
+*Reference: [Settings Tutorial in IntelliJ SDK Docs][docs:settings_tutorial]*
+
+## Quickstart
+
+This project illustrates a custom Application-level Settings through the implementation of:
```
Comment (JohnHake) [inline code_samples/settings/README.md:6]: Suggest: _This project illustrates creating custom Application-level Settings..._

File: code_samples/simple_language_plugin/src/main/gen/org/intellij/sdk/language/SimpleLexer.java [hunk 10]
```diff
@@ -11,7 +11,7 @@
 
 /**
  * This class is a scanner generated by 
- * <a href="http://www.jflex.de/">JFlex</a> 1.7.0
+ * <a href="https://www.jflex.de/">JFlex</a> 1.7.0
```
Comment (JohnHake) [inline code_samples/simple_language_plugin/src/main/gen/org/intellij/sdk/language/SimpleLexer.java:14]: Please don't edit generated files. This looks like a link embedded in a JavaDoc comment written by the jflex tool.

File: code_samples/theme_basics/README.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
+# Theme Basics [![JetBrains IntelliJ Platform SDK Docs](https://jb.gg/badges/docs.svg)][docs]
+*Reference: [Creating Custom UI Themes in IntelliJ SDK Docs][docs:themes]*
+
+## Quickstart
+
+Custom UI Themes are available beginning in version 2019.1.
+
+Creating a custom UI Theme is a process of choosing a base IDE Theme (Light or Darcula) then changing aspect
+of the base Theme definition. Custom UI Themes can:
```
Comment (JohnHake) [inline code_samples/theme_basics/README.md:9]: Suggest: _Creating a custom UI Theme is a process of choosing a base IDE Theme (Light or Darcula,) then changing aspects of the base Theme definition._

File: code_samples/theme_basics/README.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+- alter the borders and insets of UI controls,
+- provide custom editor schemes,
+- add background images.
+
+## Structure
+
+Theme Basics
+plugin depends on the [IntelliJ Platform SDK][docs] and [DevKit][docs:devkit] as a build system.
+
+The main plugin definition file is stored in the [plugin.xml][file:plugin.xml] file, which is created accordingly
```
Comment (JohnHake) [inline code_samples/theme_basics/README.md:21]: "accordingly" may be correct (?) but I would use "according" here.

File: code_samples/tool_window/README.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
+# Tool Window Sample [![JetBrains IntelliJ Platform SDK Docs](https://jb.gg/badges/docs.svg)][docs]
+*Reference: [Tool Windows in IntelliJ SDK Docs][docs:tool_windows]*
+
+## Quickstart
+
+Tool Windows are child windows of the IDE used to display information. These windows generally have their own toolbars
+(referred to as tool window bars) along the outer edges of the main window containing one or more tool window buttons,
+which activate panels displayed on the left, bottom, and right sides of the main IDE window.
+
+Current implementation displays a `JPanel` component containing simple icons and information regarding the actual
```
Comment (JohnHake) [inline code_samples/tool_window/README.md:10]: Suggest: _The current implementation displays a `JPanel` component containing simple icons and information about the actual..._

File: code_samples/tree_structure_provider/README.md [hunk 14]
```diff
@@ -0,0 +1,28 @@
+# Tree Structure Provider Sample [![JetBrains IntelliJ Platform SDK Docs](https://jb.gg/badges/docs.svg)][docs]
+*Reference: [Tree Structure View in IntelliJ SDK Docs][docs:tree_structure_view]*
+
+## Quickstart
+
+Tree Structure Provider sample project implements `treeStructureProvider` Extension Point, which allows modifying
```
Comment (JohnHake) [inline code_samples/tree_structure_provider/README.md:6]: Suggest: _The Tree Structure Provider sample project..._

File: code_samples/tree_structure_provider/README.md [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+the structure of the project tree in the Project View panel. This implementation limits the presented files
+to the Plain Text files only.
+
+Current implementation checks if a Project View nodes represents directories or file of the `PlainTextFileType` type.
```
Comment (JohnHake) [inline code_samples/tree_structure_provider/README.md:10]: Suggest: _The current implementation..._

File: code_samples/README.md [hunk 16]
```diff
@@ -0,0 +1,49 @@
+# IntelliJ Platform SDK Code Samples
```
[the remark this answers is not in the document]
Comment (hsz) [reply code_samples/README.md:1]: In general, he shouldn't even touch the `code_samples` directory and use it as it is.

File: code_samples/conditional_operator_intention/README.md [hunk 17]
```diff
@@ -0,0 +1,65 @@
+# Conditional Operator Converter [![JetBrains IntelliJ Platform SDK Docs](https://jb.gg/badges/docs.svg)][docs]
```
[the remark this answers is not in the document]
Comment (hsz) [reply code_samples/conditional_operator_intention/README.md:1]: The name is taken from `plugin.xml` file:

```
<name>SDK: Conditional Operator Converter</name>
```

Should we rename both plugin.xml and README header then?

File: code_samples/theme_basics/README.md (hunk 11 above)
Comment (hsz) [reply code_samples/theme_basics/README.md:9]: Is this comma: `(Light or Darcula,) then` written by purpose?

**Comment by hsz** [comment]
> * One sentence per line in Markdown files pls. (Except in tables, etc.)
Do you mean to don't break the lines manually?

> * Let's talk about squashing commits before opening PRs.
I'll definitely do the _Squash and merge_ - there is no need to keep this amount of commits in the main branch.

**Comment by hsz** [comment]
@JohnHake Thanks again for the review!

> * One sentence per line in Markdown files pls. (Except in tables, etc.)

Do you mean to don't break the lines manually?

> * Let's talk about squashing commits before opening PRs.

I'll definitely do the _Squash and merge_ - there is no need to keep this amount of commits in the main branch.

**Pull request merged**